### ISSUE TITLE
Allow to name wallets

### ISSUE THREAD

**Issue by AW-STJ** [AlphaWallet/alpha-wallet-android#1665]
We have implemented the same functionality for Qoin. Let us get that into AlphaWallet as well.

**Comment by JamesSmartCell** [comment]
Note that most of the code is in place (as we used to support it) but the names showing were removed for the Fiji upgrade.

Check with @colourfreak the status of this before starting on it.

**Comment by ChintanRathod** [comment]
@JamesSmartCell OK. I will coordinate with him.

**Comment by colourfreak** [comment]
@AW-STJ I don't have access to Qoin. It requires login etc.

**Comment by AW-STJ** [comment]
Will connect over Telegram to share the builds and access

**Comment by colourfreak** [comment]
<img width="1281" alt="Screenshot 2020-11-10 at 09 32 27" src="https://user-images.githubusercontent.com/51817359/98649232-a7c7e180-2337-11eb-88cb-175c994ec3e4.png">

https://zpl.io/b6qDOJy
https://zpl.io/aNrDdEe
https://zpl.io/awv69n1

**Comment by ChintanRathod** [comment]
@colourfreak 1 design is pending where list of Wallets are displayed. Could you also provide the same. 

![image](https://user-images.githubusercontent.com/4371780/98667962-2d1db700-2375-11eb-9b39-30c554ad62a3.png)

**Comment by ChintanRathod** [comment]
@AW-STJ 

What to priotarize like there is also a ENS name which is auto populated if found, so it will clash with Wallet name.

**Comment by JamesSmartCell** [comment]
Hi @colourfreak this is main AW not Qoin, which already has this ability.

With Fiji, we removed the ability to name wallets. I believe @AW-STJ has a requirement that people have contacted him wanting to re-add this ability.

I don't know which is best, just being able to edit the ENS name with a custom one, or a drop-down style selection menu or ENS taking priority. I am thinking this confusion is why we removed the wallet naming in the first place.

@ChintanRathod 

I have been adding the components necessary for creating the updated views in #1657, as per the spec in Zeplin.

This mod is at a bad time because there is big risk in duplicating work. Please leave this one until the spec is formalised.

**Comment by ChintanRathod** [comment]
@JamesSmartCell Ok Got the point. As discussed, I am holding the work on this ticket as of now.

**Comment by colourfreak** [comment]
Update: Add an additional item in the Settings tab. This just a local name. You can choose your own, disregarding the ENS name set for this wallet.

<img width="924" alt="Screenshot 2020-11-11 at 09 04 02" src="https://user-images.githubusercontent.com/51817359/98786325-3d2da900-23fe-11eb-8dd2-7bcfd979ff0e.png">

**Comment by colourfreak** [comment]
https://zpl.io/VY7JKYk
https://zpl.io/V48oWBX
https://zpl.io/29rnWDd
https://zpl.io/V0Kp3QR

**Comment by JamesSmartCell** [comment]
Thanks for the update. 

The other thing that would be good is to be able to swap ENS names for a wallet. Some ppl have multiple ENS names for the same wallet, would be good to be able to type a different one in and get that. However, it's very low priority. In fact you could achieve that with your mod although it's not technically checking the ENS.

I guess you could tie these wallet names to the address book maybe? For transferring between accounts.

**Comment by JamesSmartCell** [comment]
@ChintanRathod please resume work, but move to the Component Upgrade branch. I added details to the existing PR outlining how to cope with ENS names vs user wallet names, and only allowing names that don't appear to be valid ENS names.

**Comment by JamesSmartCell** [comment]
@ChintanRathod One further point. After consideration for complexity - in the wallet class can you remove the 'name' member and rename 'ENSname' to 'name'. This way, it should make operation a bit more clear. Can you also clear the field from the relevant Realm entity and add the migration.

**Comment by ChintanRathod** [comment]
@JamesSmartCell We have done with the integration after necessary changes. This issue could be closed.
